compiler/semstmts.nim

```diff
@@ -1106,6 +1106,22 @@ proc typeSectionFinalPass(c: PContext, n: PNode) =
         # type aliases are hard:
         var t = semTypeNode(c, x, nil)
         assert t != nil
+        if t.kind == tyGenericBody:
+          var tgi = newTypeS(tyGenericInvocation, c)
```

Comment:
At the very least move it to a helper proc like ``repairGenericAliases``.

Reply:
There is some very similar code in `liftParamType` (defined in `semtypes.nim`).
I think you can extract a shared helper proc.